source/adapters/level_zero/queue.cpp

```diff
@@ -1207,16 +1207,16 @@ ur_queue_handle_t_::executeCommandList(ur_command_list_ptr_t CommandList,
         (ZeCommandQueue, 1, &ZeCommandList, CommandList->second.ZeFence));
     if (ZeResult != ZE_RESULT_SUCCESS) {
       this->Healthy = false;
-      if (ZeResult == ZE_RESULT_ERROR_UNKNOWN) {
-        // Turn into a more informative end-user error.
-        return UR_RESULT_ERROR_UNKNOWN;
-      }
       // Reset Command List and erase the Fence forcing the user to resubmit
       // their commands.
       std::vector<ur_event_handle_t> EventListToCleanup;
       resetCommandList(CommandList, true, EventListToCleanup, false);
       CleanupEventListFromResetCmdList(EventListToCleanup,
                                        true /* QueueLocked */);
+      if (ZeResult == ZE_RESULT_ERROR_UNKNOWN) {
+        // Turn into a more informative end-user error.
+        return UR_RESULT_ERROR_UNKNOWN;
```

Comment:
If we are moving this branch down to here, wouldn't the `return ze2urResult(ZeResult)` function right below this line do exactly this same error conversion for return value (based on the switch statement inside ze2urResult)?

Reply:
It looks like that line was added afterwards, yes that branch is no longer needed, I will remove.

Reply:
Resolved, I have removed the if entirely.